internal/runners/checkout/checkout.go

```diff
@@ -77,30 +77,35 @@ func NewCheckout(prime primeable) *Checkout {
 }
 
 func (u *Checkout) Run(params *Params) (rerr error) {
+	var ns project.Namespaced
 	var archive *checkout.Archive
-	if params.FromArchive != "" {
+
+	if strings.HasSuffix(params.Namespace, checkout.ArchiveExt) {
 		var err error
-		archive, err = checkout.NewArchive(params.FromArchive)
+		archive, err = checkout.NewArchive(params.Namespace)
 		if err != nil {
 			return errs.Wrap(err, "Unable to read archive")
 		}
 		defer archive.Cleanup()
-		params.Namespace = archive.Namespace
+		ns = *archive.Namespace
```

Comment:
```suggestion
		ns = archive.Namespace
```

Doesn't seem work dereferencing when all your use of it doesn't seem to care, and in fact you create a reference again when passing it to `checkout.Run()`. May as well avoid the potential nil pointer.

Reply:
`ns` cannot be a pointer because of the `ns.Set()` function used earlier. The dereference here is needed, as is the passing of `&ns` to `checkout.Run()`. It's a product of our clever use the first argument being either a namespace or archive file.